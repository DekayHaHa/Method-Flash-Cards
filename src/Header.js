import React, { Component } from 'react';

export default class Header extends Component {
	constructor(props) {
		super(props);
		this.state = {}
	}
	detectReset = (e) => {
		const results = e.target.className.includes('round') ? false : true;
		this.props.reset(results);
		this.props.clear();
		if (results) this.props.clearAll()
	}
	render() {
		let {incorrect, correct, check} = this.props
		let checkProgess = correct + incorrect
		return (
			<nav>
				<button disabled={check} onClick={this.detectReset} className='round header-button'>Do Better</button>
				<div>
					<h2>Methods of MADNESS</h2>
					<div className='score-board'>
					<p>Correct: {correct}</p> <p>Incorrect: {incorrect}</p>
					</div>
				</div>
				<button disabled={checkProgess === 0} onClick={this.detectReset} className='all header-button'>Reset All</button>
			</nav>
		);
	}
}