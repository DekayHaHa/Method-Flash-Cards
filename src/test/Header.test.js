import React from 'react';
import ReactDOM from 'react-dom';
import Header from '../Header';
import { shallow } from 'enzyme';

const oneMock = jest.fn();
const twoMock = jest.fn();
const threeMock = jest.fn();
const mockEvent = { target: { className: 'all header-button'}}
describe('Header', () => {
	let wrapper;
	beforeEach(() => {
		wrapper = shallow(
			<Header check={true} reset={threeMock} clear={oneMock} correct={2} incorrect={3} clearAll={twoMock} />
		);
	});
	it('should match snapshop', () => {
		expect(wrapper).toMatchSnapshot();
	});
	it('should have default state', () => {
		expect(wrapper.state()).toEqual({
		});
	});
	it('should be able to reset round', () => {
		wrapper.instance().detectReset(mockEvent)
		expect(oneMock).toBeCalled();
		expect(threeMock).toBeCalled();
		expect(twoMock).toBeCalled();
	})
})