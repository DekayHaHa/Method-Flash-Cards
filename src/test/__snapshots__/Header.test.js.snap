// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshop 1`] = `
<nav>
  <button
    className="round header-button"
    disabled={true}
    onClick={[Function]}
  >
    Do Better
  </button>
  <div>
    <h2>
      Methods of MADNESS
    </h2>
    <div
      className="score-board"
    >
      <p>
        Correct: 
        2
      </p>
       
      <p>
        Incorrect: 
        3
      </p>
    </div>
  </div>
  <button
    className="all header-button"
    disabled={false}
    onClick={[Function]}
  >
    Reset All
  </button>
</nav>
`;
